tools/plugin_sync/plugin_sync.pt

```diff
@@ -0,0 +1,200 @@
+      field "file_name", jmes_path(col_item, "file_name")
+      field "version", jmes_path(col_item, "version")
+      field "change_log", jmes_path(col_item,"change_log")
+    end
+  end
+end
+
+#get the list of published plugins
+datasource "ds_published_plugin_list" do
+  request do
+    auth $auth_rs
+    verb "GET"
+    host 'cloud-3.rightscale.com'
+    path join(["/cloud/v1/orgs/",rs_org_id,"/plugins"])
+    header "Api-Version", "1.0"
+    #query "view", "extended"
+  end
+  result do
+    collect jmes_path(response,"items") do
+      field "name",     jmes_path(col_item, "name")
+      field "file_name", jmes_path(col_item, "fileName")
+      field "info",     jmes_path(col_item, "info")
+      field "version",  jmes_path(col_item, "version")
+    end
+  end
+end
+
+datasource "ds_combined_plugin_list" do
+  run_script $js_combine_plugin_lists, $ds_active_plugin_list, $ds_published_plugin_list, $force_upgrade
+end
+
+script "js_combine_plugin_lists", type: "javascript" do
+  parameters "active_plugin_list", "published_plugin_list", "force_upgrade"
+  result "plugin_list"
+  code <<-EOS
+  var plugin_list = [];
+  var old_plugin_list = {};
+  var github_raw_url = 'https://raw.githubusercontent.com/flexera/flexera-plugins/master/';
+
+  // create a map of published templates
+  for (var index = 0; index < published_plugin_list.length; index++) {
+    var published_plugin = published_plugin_list[index];
+    old_plugin_list[published_plugin.name] = published_plugin;
+  }
+
+  // loop through active plugins from s3 json file
+  for ( var i = 0; i < active_plugin_list.length; i++) {
+    var active_plugin = active_plugin_list[i]
+    // map catalog plugin with json list
+    var account_plugin = old_plugin_list[active_plugin.name]
+
+    var plugin_name = ""
+    var account_version = "0.0.0"
+    if (account_plugin){
+      var account_version = account_plugin.version
+      var plugin_name = account_plugin.name
+    }
+    // published version
+    var a_version = account_version.toString().split('.')
+    // repo version
+    var r_version = active_plugin.version.toString().split('.')
+
+    var version_upgradeable = false
+    curr_major_version = parseInt(a_version[0])
+    curr_minor_version = parseInt(a_version[1])
+    curr_patch_version = parseInt(a_version[2])
+    new_major_version = parseInt(r_version[0])
+    new_minor_version = parseInt(r_version[1])
+    new_patch_version =parseInt(r_version[2])
+
+    if (curr_major_version < new_major_version ) {
+      var version_upgradeable = true
+    } else if (curr_minor_version < new_minor_version) {
+        var version_upgradeable = true
+    } else if (curr_patch_version < new_patch_version) {
+        var version_upgradeable = true
+    }
+
+    if (force_upgrade == 1 ) {
+      var version_upgradeable = true
+    }
+
+    plugin_list.push({
+      name: active_plugin.name,
+      file_name: active_plugin.file_name,
+      repo_version: active_plugin.version,
```

Comment:
should we name this plugin_version ?

Reply:
There are two plugin version, one is the repo and one in the catalog.